workcal/angular/src/app/calendar-page/calendar-page.component.ts

```diff
@@ -0,0 +1,482 @@
+      this.fetchEvents();
+    },
+    (error) => {
+      console.error('Error updating event:', error);
+    }
+
+  );
+  this.fetchEvents();
+  this.refreshPage();
+
+  }
+
+  deleteEvent(eventId: string): void {
+    const index = this.events.findIndex(event => event.id === eventId);
+    if (index !== -1) {
+      this.eventApiService.deleteEvent(eventId).subscribe(() => {
+        this.events.splice(index, 1);
+        this.fetchEvents();
+      });
+    }
+    this.refreshPage();
+
+  }
+
+  onEventAdding(event): void {
+    const appointmentData = event.appointmentData;
+
+    const selectedLabels = this.defaultLabels.filter(label =>
+      appointmentData.labels?.includes(label.name)
+    );
+
+      const selectedUsers = appointmentData.users.filter(user =>
+        appointmentData.users?.includes(user.name)
+      );
+
+      const selectedUserIDs = appointmentData.users;
+
+
+
+    const newEvent: EventDto = {
+      name: appointmentData.text,
+      startTime: new Date(appointmentData.startDate),
+      endTime: new Date(appointmentData.endDate),
+      location: appointmentData.location || '',
+      labels: selectedLabels,
+      users:  selectedUserIDs
+    };
+
+    this.createEvent(newEvent);
+    this.fetchEvents();
+    this.refreshPage();
+
+  }
+
+
+  onAppointmentDragStart(event): void {
+    this.draggedEventLabels = event.itemData.labels;
+
+  }
+
+  onAppointmentDragEnd(event): void {
+    if (event.fromComponent === event.toComponent) {
+      const appointmentData = event.itemData;
+      appointmentData.labels = this.draggedEventLabels;
+      this.updateEvent(appointmentData);
+    }
+    this.fetchEvents();
+    this.refreshPage();
+
+  }
+
+  onEventUpdating(event): void {
+
+    //console.log("Event data before any operation: ", JSON.stringify(event));
+
+    const appointmentData = event.newData;
+    const appointmentDataOld = event.oldData;
+    console.log('New appointmentData:', appointmentData);
+    console.log('Initial appointmentData.users:', appointmentData.users);
+
+    if (!appointmentDataOld.id) {
+      console.error("Event ID is missing, cannot update");
+      return;
+    }
+
+    appointmentData.labels = appointmentData.labels || [];
+    appointmentDataOld.labels = appointmentDataOld.labels || [];
+
+    let selectedLabels = [];
+
+    if (this.labelsInteractedWith) {
+      selectedLabels = this.defaultLabels.filter(label => appointmentData.labels.includes(label.name));
+    } else {
+      selectedLabels = appointmentDataOld.labels;
+    }
+
+
+    appointmentData.users = appointmentData.users.map(userId => {
+      return this.allusers.find(user => user.id === userId) || userId;
+    });
+
+    console.log('New appointmentData:', appointmentData.users);
+
+    let selectedUserIDs: UserDto[] = [];
+
+    // Check the type of the first element in the array
+    if (Array.isArray(appointmentData.users) && appointmentData.users.length > 0) {
```

Comment:
Should we store liek this different state of a resource?
Easier to maintain if we store in different variable these values then using one allocation into different purposes.